scripts/openwebnet-discovery.nse

```diff
@@ -137,10 +137,15 @@ local function get_response(sd, request)
 
     if status == nil then
       stdnse.debug("Error: " .. tostring(data))
-      -- Captures the NACK after TIMEOUT occured.
-      -- Avoids false results.
-      status, data = sd:receive_buf("##", true)
-      break
+      if tostring(data) == "TIMEOUT" then
```

Comment:
Incase we receive `nil` as an error due to some reason, converting `data` to string won't raise throw any error during printing and comparing.